crate/utils/src/crypto/hybrid_encryption/encryption.rs

```diff
@@ -76,15 +71,30 @@ impl HybridEncryptionSystem {
 impl EncryptionSystem for HybridEncryptionSystem {
     fn encrypt(&self, request: &Encrypt) -> Result<EncryptResponse, KmipUtilsError> {
         let plaintext = request.data.clone().context("missing plaintext data")?;
-        let mut rng = self.rng.lock().unwrap();
+
+        #[cfg(not(feature = "fips"))]
+        let rng = Arc::new(Mutex::new(CsRng::from_entropy()));
+        #[cfg(not(feature = "fips"))]
+        let mut rng = rng.lock().expect("RNG lock poisoned");
 
         // Convert the Pkey to a crypto_core curve and perform emcryption
         // Note: All conversions below will go once we move to full openssl
         let id = self.public_key.id();
         let ciphertext: Vec<u8> = match id {
-            Id::EC => encrypt_with_nist_curve(&mut rng, &self.public_key, &plaintext)?,
+            Id::EC => ecies_encrypt(&self.public_key, &plaintext)?,
+            Id::RSA => {
+                if self.key_wrapping {
+                    ckm_rsa_aes_key_wrap(self.public_key.clone(), &plaintext)?
+                } else {
+                    rsa_oaep_aes_gcm_encrypt(
+                        self.public_key.clone(),
```

Comment:
Why do you need ownership here?

Reply:
refer to commit 583918e08e78aba4b91897e711b2305e233e650a for change